en/docs/visual-programming/data-mapper.md

```diff
@@ -1 +1,160 @@
+
+>**Info:** You can use Ballerina query support to convert one array type to another. To use a query in a Data Mapper, you can select the array by clicking on it. Then, it will provide you with several buttons. Click the code action button (bulb sign) and select **Convert to query**. Then, the Data Mapper will convert the mapping to a query. Then, move into the query and do the mapping between the array types.
+
+![Convert to Query](img/dm-convert-to-query.gif "Map via query expression")
+
+#### Further process the data
+You can further process the data within the query expression. Currently, the Data Mapper supports the following intermediate clauses.
+- Where clause: filter data based on a given condition
+- Let clause: define local variables within the query expression
+- Limit clause: limit the number of elements returns from the query expression
+- Order by clause: sort data within the query expression in `ascending` or `descending` directions
+- Join clause: performs an inner or left outer join
+- Outer join clause: performs an outer join
+
+![Further Processing](img/dm-further-processing.gif "Process via intermediate clauses")
+
+Once array type mapping is completed, select the transform function name in the top breadcrumb bar to navigate to the root view of mapping.
+
+### Add local variables
+You can define local variables within the transformation function and re-use in multiple places to avoid duplications.
+>**Info:** If there are no existing local variables, you can open up the local variable pane by clicking on the `Add Local Variable` button. Otherwise, the defined local variables are listed down under local variables in the RHS of the UI. You will be able to access the local variable pane by clicking on the edit button there.
+
+![Local Variable](img/dm-add-local-variable.gif "Add local variable")
+
+### Manipulate fields without drawing connections
+Click on the triple dots button at the end of the field to see the actions that can be performed for the selected field.
+
+#### Initialize arrays, Add and delete elements
+>**Info:** The actions are provided based on the type of the selected field. If you have click on the triple dots button of an array typed field, you will see `Initialize Array` option.
+
+Once the array is initialized, you can add array elements by clicking on the `+ Add Element` button.
+Furthermore, clicking on the triple dots button on any array element will provide option to delete that element.
+
+![Array Manipulation](img/dm-array-manipulation.gif "Manipulate arrays")
+
+#### Add/Edit constant values/expressions 
+>**Info:** If a particular field is empty and accepting constant/expression you will see `Add value` action after clicking on the triple dots button. Click on it will open up the expression editor and there you will be able to provide a constant value or construct any complex expression.
```

Comment:
```suggestion
>**Info:** If a particular field is empty and accepting a constant/expression, you will see the **Add value** action after clicking the triple dots button. This will open up the expression editor and you will be able to provide a constant value or construct any complex expression in it.
```